lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -0,0 +1,1300 @@
+          launch_group:
+          wait_timeout: 300
+    '''
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        state=dict(default='present', choices=['present', 'started', 'running', 'stopped', 'restarted', 'rebooted', 'terminated', 'absent']),
+        wait=dict(default=True, type='bool'),
+        wait_timeout=dict(default=600, type='int'),
+        # count=dict(default=1, type='int'),
+        assign_public_ip=dict(type='bool'),
+        image=dict(type='dict'),
+        image_id=dict(type='str'),
+        instance_type=dict(default='t2.micro', type='str'),
+        user_data=dict(type='str'),
+        tower_callback=dict(type='dict'),
+        ebs_optimized=dict(type='bool'),
+        vpc_subnet_id=dict(type='str', aliases=['subnet_id']),
+        security_groups=dict(default=[], type='list'),
+        security_group=dict(type='str'),
+        instance_role=dict(type='str'),
+        name=dict(type='str'),
+        tags=dict(type='dict'),
+        purge_tags=dict(type='bool', default=False),
+        filters=dict(type='dict', default=None),
+        instance_initiated_shutdown_behavior=dict(type='str', choices=['stop', 'terminate']),
+        termination_protection=dict(type='bool'),
+        instance_ids=dict(default=[], type='list'),
+        network=dict(default=None, type='dict'),
+    ))
+    # running/present are synonyms
+    # as are terminated/absent
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['network_interfaces', 'network'],
+            ['security_groups', 'security_group'],
+            ['network_interfaces', 'security_group'],
+            ['network_interfaces', 'security_groups'],
+            ['tower_callback', 'user_data'],
+            ['image_id', 'image'],
+        ],
+        supports_check_mode=True
+    )
+
+    def client(self, service):
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(self, boto3=True)
+        return boto3_conn(self, conn_type='client', resource=service,
+                          region=region, endpoint=ec2_url, **aws_connect_kwargs)
+
+    def resource(self, service):
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(self, boto3=True)
+        return boto3_conn(self, conn_type='resource', resource=service,
+                          region=region, endpoint=ec2_url, **aws_connect_kwargs)
+
+    module.client = lambda x: client(module, x)
+    module.resource = resource
+
+    state = module.params.get('state')
+    ec2 = module.client('ec2')
+    if module.params.get('filters') is None:
+        filters = {
+            # all states except shutting-down and terminated
+            'instance-state-name': ['pending', 'running', 'stopping', 'stopped']
+        }
+        if state == 'stopped':
+            # only need to change instances that aren't already stopped
+            filters['instance-state-name'] = ['pending', 'running']
```

Comment:
If they want to wait for an instance that is 'stopping' but not 'stopped' should that be possible?

Reply:
Yeah, we can make that a thing. Implemented.